components/eventing-controller/pkg/handlers/nats_test.go

```diff
@@ -180,6 +181,107 @@ func TestSubscription(t *testing.T) {
 	}
 }
 
+func TestMultipleSubscriptionsToSameEvent(t *testing.T) {
+	g := NewWithT(t)
+	natsPort := 5222
+	subscriberPort := 8080
+	subscriberReceiveURL := fmt.Sprintf("http://127.0.0.1:%d/store", subscriberPort)
+	subscriberCheckURL := fmt.Sprintf("http://127.0.0.1:%d/check", subscriberPort)
+
+	// Start Nats server
+	natsServer := eventingtesting.RunNatsServerOnPort(natsPort)
+	defer natsServer.Shutdown()
+
+	defaultLogger, err := logger.New(string(kymalogger.JSON), string(kymalogger.INFO))
+	if err != nil {
+		t.Fatalf("initialize logger failed: %v", err)
+	}
+
+	natsConfig := env.NatsConfig{
+		Url:           natsServer.ClientURL(),
+		MaxReconnects: 2,
+		ReconnectWait: time.Second,
+	}
+	defaultMaxInflight := 1
+	natsClient := NewNats(natsConfig, env.DefaultSubscriptionConfig{MaxInFlightMessages: defaultMaxInflight}, defaultLogger)
+
+	if err := natsClient.Initialize(env.Config{}); err != nil {
+		t.Fatalf("connect to Nats server failed: %v", err)
+	}
+
+	// Create a new subscriber
+	subscriber := eventingtesting.NewSubscriber(fmt.Sprintf(":%d", subscriberPort))
+	subscriber.Start()
+
+	// Shutting down subscriber
+	defer subscriber.Shutdown()
+
+	// Check if the subscriber is running or not by checking its store
+	err = subscriber.CheckEvent("", subscriberCheckURL)
+	if err != nil {
+		t.Fatalf("subscriber did not receive the event: %v", err)
+	}
+
+	// Prepare event-type cleaner
+	application := applicationtest.NewApplication(eventingtesting.ApplicationNameNotClean, nil)
+	applicationLister := fake.NewApplicationListerOrDie(context.Background(), application)
+	cleaner := eventtype.NewCleaner(eventingtesting.EventTypePrefix, applicationLister, defaultLogger)
+
+	// Create 3 subscriptions having the same sink and the same event type
+	var subs [3]*eventingv1alpha1.Subscription
+	for i:=0; i< len (subs); i++ {
+		subs[i] = eventingtesting.NewSubscription("sub" + strconv.Itoa(i), "foo", eventingtesting.WithNotCleanEventTypeFilter)
```

Comment:
nit: `fmt` is already imported, so 
```
fmt.Sprintf("sub-%d", i)
```
```suggestion
		subs[i] = eventingtesting.NewSubscription(fmt.Sprintf("sub-%d", i), "foo", eventingtesting.WithNotCleanEventTypeFilter)
```

Reply:
Done